src/base/bittorrent/torrentinfo.cpp

```diff
@@ -213,24 +214,79 @@ QByteArray TorrentInfo::metadata() const
 
 QStringList TorrentInfo::filesForPiece(int pieceIndex) const
 {
-    if (pieceIndex < 0)
-        return QStringList();
+    // no checks here since fileIndicesForPiece() will return an empty list
+    QList<int> fileIndices {fileIndicesForPiece(pieceIndex)};
+
+    QStringList res;
+    std::transform(fileIndices.begin(), fileIndices.end(), std::back_inserter(res), [this](int i) {
+        return this->filePath(i);
+    });
+
+    return res;
+}
+
+QList<int> BitTorrent::TorrentInfo::fileIndicesForPiece(int pieceIndex) const
+{
+    if (!isValid() || ( pieceIndex < 0) || ( pieceIndex >= piecesCount()) )
+        return QList<int>();
 
     std::vector<libtorrent::file_slice> files(
         nativeInfo()->map_block(pieceIndex, 0, nativeInfo()->piece_size(pieceIndex)));
-    QStringList res;
-    for (const libtorrent::file_slice& s: files) {
-        res.append(filePath(s.file_index));
-    }
+    QList<int> res;
+    std::transform(files.begin(), files.end(), std::back_inserter(res), [](const libtorrent::file_slice &s) {
+        return s.file_index;
+    });
+
     return res;
 }
 
+QPair<int, int> BitTorrent::TorrentInfo::piecesForFile(const QString& file) const
+{
+    if (!isValid()) // if we do not check here the debug message will be printed, which would be not correct
+        return qMakePair(-1, -1);
+
+    int index = fileIndex(file);
+    if (index == -1) {
+        qDebug() << "Filename '" << file << "' was not found in torrent '" << name() << '\'';
+        return qMakePair(-1, -1);
+    }
+    return piecesForFile(index);
+}
+
+QPair<int, int> BitTorrent::TorrentInfo::piecesForFile(int fileIndex) const
+{
+    if (!isValid())
+        return qMakePair(-1, -1);
+
+    if (( fileIndex < 0) || ( fileIndex >= filesCount()) ) {
+        qDebug() << "File index (" << fileIndex << ") is out of range for torrent '" << name() << '\'';
+        return qMakePair(-1, -1);
+    }
+
+    const auto &files = nativeInfo()->files();
```

Comment:
Get rid of these **auto**s. We make only limited use of **auto** keyword. See CODING_GUIDELINES.md